### PR TITLE
Combine ArrayChunks by default

### PR DESCRIPTION
**Pull request by ddn0**
From microbenchmark results, simply having the code path to support a
sequence of arrow::Arrays but never using it, resulted in a 4X slowdown
relative to simple pointer accesses when traversing a large random
graph. It also inhibited vectorization.

Since our main use case is random access within a column (e.g., get node
data for a neighbor), always return a combined column so that downstream
code can optimize specifically for this case.

Microbenchmark Results
======================

Iterate over a random graph (out degree: 4, number of nodes: 2^20,
property type: int64_t) with 3 edge and node properties, each an
arrow::ChunkedArray of size 1. For each node, sum edge and node data
from neighbors.
- Baseline: manually extract pointers from the arrow::Array and assume
  that there is only one chunk in the arrow::ChunkedArray
- StaticBaseline: Baseline with number of properties determined
  statically
- Chunk: A templatized typed iterator that performs all the work to
  support random access over ChunkedArrays (e.g., maintain current
  array, check if we've reached the end of the array) but, for the
  benchmark, modify it to simply do basic pointer operations like
  Baseline
- ChunkInline: Like Chunk but explicitly inline as false "have we
  reached the end of the array" checks
```
----------------------------------------------------------------------------
Benchmark                         Time             CPU Iterations
----------------------------------------------------------------------------
Baseline/1048576           93093403 ns     93092933 ns 8
StaticBaseline/1048576     74204233 ns     74203053 ns 9
Chunk/1048576/3/0         375272142 ns    375264067 ns 2
ChunkInline/1048576/3/0    73680304 ns     73679838 ns 9
```
ChunkedInline and StaticBaseline are faster than Baseline because the
compiler can vectorize parts of the traversal because the number and
type of properties is statically known. For much larger inputs (32X),
the benefit of vectorization disappears.

### CODE REVIEW

**Comment by ddn0** [comment]
I didn't include the benchmark itself because there is no natural place for it right now. My plan is to include some form of this baseline versus typed iterator benchmark with the work that implements the new property graph iterator.

**Comment by ddn0** [comment]
There's binary (int32_t offsets) and large binary (int64_t) arrow types. If we can convert them to large binary, we wouldn't need to support multichunked columns.

**Comment by l-hoang** [comment]
I haven't looked at the code yet (will do in a bit), but if I recall correctly there was a limit on the size that a single chunk could be. Does this PR assume that everything can always be in a single chunk?

File: libtsuba/src/RDG.cpp [hunk 1]
```diff
@@ -190,6 +190,19 @@ DoLoadTable(const std::string& expected_name, const std::string& file_path) {
     return tsuba::ErrorCode::ArrowError;
   }
 
+  // Combine multiple chunks into one. Binary and string columns (c.f. large
+  // binary and large string columns) are a special case. They may not be
+  // combined into a single chunk due to the fact the offset type for these
+  // columns is int32_t and thus the maximum size of an arrow::Array for these
+  // types is 2^31.
```
Comment (l-hoang) [inline libtsuba/src/RDG.cpp:197]: This answers my question.

**Comment by roshandathathri** [comment]
Checking this now but feel free to merge

**Pull request merged**